File: library/src/main/java/lib/virgo/library/util/ScrollListenerAdapter.java
package lib.virgo.library.util;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import lib.virgo.library.ScrollToCenterView;

/**
 * Rebuild from DroidSky/DiscreteScrollView
 * <a href="https://github.com/DroidSky/DiscreteScrollView">...</a>
 */
public class ScrollListenerAdapter<T extends RecyclerView.ViewHolder> implements ScrollToCenterView.ScrollStateChangeListener<T> {

    private ScrollToCenterView.ScrollListener<T> adaptee;

    public ScrollListenerAdapter(ScrollToCenterView.ScrollListener<T> adaptee) {
        this.adaptee = adaptee;
    }

    @Override
    public void onScrollStart(@NonNull T currentItemHolder, int adapterPosition) {

    }

    @Override
    public void onScrollEnd(@NonNull T currentItemHolder, int adapterPosition) {

    }

    @Override
    public void onScroll(float scrollPosition, @NonNull T currentHolder, @NonNull T newCurrentHolder) {
        adaptee.onScroll(scrollPosition, currentHolder, newCurrentHolder);
    }
}

File: library/src/main/java/lib/virgo/lib_scrollcenteritem/util/ScrollListenerAdapter.kt
package lib.virgo.lib_scrollcenteritem.util

import androidx.recyclerview.widget.RecyclerView
import lib.virgo.lib_scrollcenteritem.callback.ScrollListener
import lib.virgo.lib_scrollcenteritem.callback.ScrollStateChangeListener

/**
 * Rebuilt from [DroidSky/DiscreteScrollView](https://github.com/DroidSky/DiscreteScrollView)
 */
class ScrollListenerAdapter<T : RecyclerView.ViewHolder>(
    private val adaptee: ScrollListener<T>,
) : ScrollStateChangeListener<T> {

    override fun onScrollStart(currentItemHolder: T, adapterPosition: Int) {}

    override fun onScrollEnd(currentItemHolder: T, adapterPosition: Int) {}

    override fun onScroll(scrollPosition: Float, currentHolder: T, newCurrent: T) {
        adaptee.onScroll(scrollPosition, currentHolder, newCurrent)
    }
}
